returnn/config.py

```diff
@@ -235,6 +236,10 @@ def __parse_python(self, code, name=None):
         if isinstance(code, str):
             return code
         if isinstance(code, DelayedBase):
+            assert self.hash_full_python_code is False, (
```

Comment:
I think this this is only necessary for `tk.Variable`. `tk.Path` should not change the hash over time.
So we _could_ allow using Path objects for now.

On the other hand, we might want to change the representation of Path/Variable in the Manager and then it would break some hashes. So maybe disallowing all `DelayedBase` objects for now is a good idea.

Thinking of it.. `tk.Path.get()` in the hash will make the hash dependent of the full path.. :/